client/packages/system/src/Name/DetailModal/DetailModal.tsx

```diff
@@ -114,14 +106,54 @@ export const DetailModal: FC<DetailModalProps> = ({ nameId }) => {
             <DetailInputWithLabelRow
               label={t('label.address')}
               inputProps={{
-                value: [data?.address1, data?.address2]
-                  .filter(a => !!a)
-                  .join(', '),
+                value: data?.address1,
                 disabled: isDisabled,
                 maxRows: 3,
                 multiline: true,
               }}
             />
+            <DetailInputWithLabelRow
+              label={t('label.address')}
+              inputProps={{
+                value: data?.address2,
+                disabled: isDisabled,
+                maxRows: 3,
+                multiline: true,
+              }}
+            />
+            <DetailInputWithLabelRow
+              label={t('label.country')}
+              inputProps={{ value: data?.country, disabled: isDisabled }}
+            />
+            <DetailInputWithLabelRow
+              label={t('label.currency')}
+              inputProps={{ value: data?.currency?.code, disabled: isDisabled }}
+            />
+            <DetailInputWithLabelRow
+              label={t('label.website')}
+              inputProps={{ value: data?.website, disabled: isDisabled }}
+              inputAlignment="start"
+              DisabledInput={
+                <MuiLink
+                  href={data.website ?? undefined}
+                  target="_blank"
+                  rel="noopener"
+                  sx={{
+                    // Make it look like another disabled text input consistent with other text
+                    // input components in this modal
+                    width: '100%',
+                    minHeight: '34.13',
+                    backgroundColor: theme => theme.palette.background.toolbar,
+                    borderRadius: '8px',
+                    padding: '4px 8px',
+                    // This is to match the width of BasicTextInput (from the required *):
+                    marginRight: '2',
+                  }}
+                >
+                  {data.website}
+                </MuiLink>
```

Comment:
Can we make this one wider? Seems unnecessary to wrap the link text when there's so much spare white space:

<img width="646" alt="Screenshot 2025-06-12 at 9 29 07 AM" src="https://github.com/user-attachments/assets/7eba47c4-624b-403e-9294-98d8422caa41" />